agent/engine/common_test.go

```diff
@@ -184,20 +184,15 @@ func addTaskToEngine(t *testing.T,
 	createStartEventsReported sync.WaitGroup) {
 	// steadyStateCheckWait is used to force the test to wait until the steady-state check
 	// has been invoked at least once
-	steadyStateCheckWait := sync.WaitGroup{}
```

Comment:
why was this removed? steadyState check should not have any bearing on wiring contexts to `waitEvent` method, right?

Reply:
I removed this because further down we do a `steadyStateCheckWait.Done()` for a `mocktime.EXPECT()...`, since this mocktime maps to this removed [code](https://github.com/aws/amazon-ecs-agent/pull/1226/files/e42e6b13f0813b3d5ce6f8115ef6751b304d46ba#diff-20346f7af6202828de353f0137b6e43aL264) in the `waitSteady()` method. I don't *think* we're losing any coverage by getting rid of this here. Let me know if you think I'm missing something.

Reply:
Since we are getting rid of the `mocktime` for `waitSteady`, how are making sure the `waitSteady` is called? Also can the line 194 be removed?

Reply:
> Also can the line 194 be removed?

We can remove it.

> how are making sure the waitSteady is called? 

Are we losing anything by not ensuring that waitSteady is being called? Since it is a private method and an implementation detail, I'm not sure we need to be explicitly checking if waitSteady is called. Please let me know what you think though.